docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,102 @@
+
+If the service finds the top-level (root) namespace is not populated with configuration information, it will read its local configuration file and populate the configuration provider (under the namespace for the service) with configuration read from the local configuration file.
+
+Note:  a configuration provider can be specified with a command line argument (the -cp / --configProvider) or environment variable (the edgex_configuration_provider environmental variable which overrides the command line argument).
+> NOTE:  the environmental variables are typically uppercase but there have been inconsistencies in environmental variable casing and changing it would involve non-backward compatible change.  This should be considered and made consistent in a future major release.
+
+**Using the local configuration file**
+
+When a configuration provider _isn't_ specified, the service just uses the configuration in its local configuration file.  That is the service uses the configuration in the file associated with the profile, config filename and config file directory command line options or environmental variables.  In this case, the service does not contact the configuration service (Consul) for any configuration information.
+  
+NOTE:  As the services now self seed and deployment specific changes can be made via environment overrides, it will no longer be necessary to have a Docker profile configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.  It will still be possible for users to use the profile mechanism to specify a Docker configuration, but it will no longer be required and not the recommended approach to providing Docker container specific configuration.
+
+### Overrides
+Environmental variables override configuration or command line option values in the following ways
+- Environmental variables override local configuration; that is when configuration for a service is obtained from the local config file
+- Environmental variables override configuration values as they are pushed (self seeded) into the configuration service (Consul).  This override only occurs once (as the values are pushed / seeded into Consul from the service).  Once pushed, configuration values are used from Consul and any environmental value is ignored unless the -o/--overwrite flag is on.  The name of the environmental variable must match the path names in Consul.
+- Environmental variables can override command line options (except the -o overwrite and -r registry command line options). 
+
+NOTES:
+- Environmental variables do not override any local configuration; that is when configuration for a service is obtained from the local config file but not from the configuration service, the environmental variables are ignored.
```

Comment:
I think we should have override of local configuration. I have implemented this in my prototype env override of local configuration.

Reply:
+1 to Lenny's comment. We discussed overriding local config in Core WG 27-Feb-2020 and sounded like this was desirable on most accounts.